spec/rmagick/image/constitute_spec.rb

```diff
@@ -1,13 +1,48 @@
 RSpec.describe Magick::Image, '#constitute' do
   let(:img) { Magick::Image.read(IMAGES_DIR + '/Button_0.gif').first }
-  let(:pixels) { img.dispatch(0, 0, img.columns, img.rows, 'RGBA') }
+  let(:pixels) { img.export_pixels(0, 0, img.columns, img.rows, 'RGBA') }
 
   it 'returns an equivalent image to the given pixels' do
-    res = Magick::Image.constitute(img.columns, img.rows, 'RGBA', pixels)
-    # The constituted image is in MIFF format so we
-    # can't compare it directly to the original image.
-    expect(res.columns).to eq img.columns
-    expect(res.rows).to eq img.rows
-    expect(pixels.all? { |v| v >= 0 && v <= Magick::QuantumRange }).to be true
+    result = Magick::Image.constitute(img.columns, img.rows, 'RGBA', pixels)
+
+    expect(result.export_pixels).to eq(img.export_pixels)
+  end
+
+  it 'allows constituting with RGBA quantum (integer) pixel values' do
+    # [R, G, B, A, R, G, B, A, ...]
+    pixels = [1] * 4 * img.columns * img.rows
+
+    result = Magick::Image.constitute(img.columns, img.rows, 'RGBA', pixels)
+
+    result_pixels = result.dispatch(0, 0, img.columns, img.rows, 'RGBA')
+    expect(result_pixels).to all eq(1)
+  end
+
+  it 'allows constituting with RGBA scale (float) pixel values' do
+    # [R, G, B, A, R, G, B, A, ...]
+    pixels = [1.0] * 4 * img.columns * img.rows
+
+    result = Magick::Image.constitute(img.columns, img.rows, 'RGBA', pixels)
+
+    result_pixels = result.dispatch(0, 0, img.columns, img.rows, 'RGBA')
+    expect(result_pixels).to all eq(65_535)
+  end
+
+  it 'raises an error when invalid RGBA pixel values are given' do
+    pixels = ['x'] * (4 * img.columns * img.rows)
+    expected_message = 'element 0 in pixel array is String, must be numeric'
+
+    expect { Magick::Image.constitute(img.columns, img.rows, 'RGBA', pixels) }
+      .to raise_error(TypeError, expected_message)
+  end
+
```

Comment:
Maybe we should also have a test for `wrong number of array elements`?

Reply:
I added a test for wrong number of array elements.